daemon/endpoint.go

```diff
@@ -538,6 +536,29 @@ func (d *Daemon) deleteEndpoint(ep *endpoint.Endpoint) int {
 	return len(errs)
 }
 
+// EndpointMonitorDeleterNotifier is a wrapper around the monitor notification
+// that a specified Endpoint has been deleted.
+type EndpointMonitorDeleterNotifier struct {
+	d *Daemon
+}
+
+// NotifyMonitorDeleted notifies the monitor that an endpoint has been deleted.
+func (emd *EndpointMonitorDeleterNotifier) NotifyMonitorDeleted(ep *endpoint.Endpoint) {
```

Comment:
Why wrap the daemon in a new structure if the daemon could implement the `NotifyMonitorDeleted` directly?